scheduler/pkg/agent/client.go

```diff
@@ -621,6 +621,21 @@ func (c *Client) LoadModel(request *agent.ModelOperationMessage, timestamp int64
 		logger.Errorf("there was a problem getting the config for model: %s", modelName)
 	}
 
+	// we are going to clean-up all the older stale version of model
+	for _, v := range c.stateManager.modelVersions.getVersionsForAllModels() {
+		// skip if version is same as current one
+		if v.GetVersion() == modelVersion {
+			continue
+		}
+		c.logger.WithField("func", c.stateManager.modelVersions.getVersionsForAllModels()).Infof("----")
+
+		err := c.ModelRepository.RemoveModelVersion(util.GetVersionedModelName(modelName, v.GetVersion()))
```

Comment:
This flow is probably not the correct flow as we are not yet sure that we should be removing the old version of this model.

In progressive rollout we have to make sure that the new version of the model is up before removing an old version otherwise we risk inference requests not being served in this transition.

I dont think this is the correct place also to do the clean up as one agent (i.e. a server replica loading one model replica) doesnt have visibility with regards to the actual state of the new version of the model (e.g. that the envoy routes have been updated). Therefore I think that the clean up process should be done somewhere at the scheduler side and not on the agent.